modules/150-user-authn/images/dex/Dockerfile

```diff
@@ -1,25 +1,18 @@
-ARG BASE_GOLANG_20_ALPINE_DEV
+ARG BASE_GOLANG_21_ALPINE_DEV
 ARG BASE_DISTROLESS
-FROM $BASE_GOLANG_20_ALPINE_DEV as artifact
+FROM $BASE_GOLANG_21_ALPINE_DEV as artifact
 ARG GOPROXY
 ENV GOPROXY=${GOPROXY}
 ARG SOURCE_REPO
 ENV SOURCE_REPO=${SOURCE_REPO}
 WORKDIR /dex
 COPY patches/client-groups.patch patches/static-user-groups.patch patches/gitlab-refresh-context.patch patches/connector-data.patch patches/oidc-ca-insecure.patch patches/robots-txt.patch patches/401-password-auth.patch /
-RUN git clone --branch v2.35.3 --depth 1 ${SOURCE_REPO}/dexidp/dex.git . \
+RUN git clone --branch v2.40.0 --depth 1 ${SOURCE_REPO}/dexidp/dex.git . \
   && git apply /client-groups.patch \
   && git apply /static-user-groups.patch \
-  && git apply /gitlab-refresh-context.patch \
-  && git apply /connector-data.patch \
-  && git apply /oidc-ca-insecure.patch \
-  && git apply /robots-txt.patch \
-  && git apply /401-password-auth.patch
```

Comment:
it would be better to delete the patches that that had made their way into v2.40.0 and update README.md accordingly.

Reply:
Completed